Update New Document Page to show document type option based on user permission & organisation.

farahTW opened the pull request:
This PR updates the New Document page for header navigation to show document type options based on user permission and organisation.
**It does the following:**

- Updates New Document Page controller and View to show and hide 'Fatality Notice' option based on User permission and Organisation.
- Adds tests to NewDocumentControllerTest to cover the scenarios.
- Adds tests to MoreControllerTest to cover the scenario for user permission and organisation.

**Screen shots:**
**Case 1:** User permission is GDS editor and Organisation is GDS: Fatality Notice is shown

![image](https://github.com/alphagov/whitehall/assets/131259138/f764e798-20f2-4ea7-a247-0a2d5a99c220)

**Case 2:** User Organisation in Ministry Of Defence irrespective of permission
![image](https://github.com/alphagov/whitehall/assets/131259138/0a3937a1-8737-404f-9d80-11f315d4663f)

**Case 3: When organisation is Cabinet Minister and no GDS Editor permission**

![image](https://github.com/alphagov/whitehall/assets/131259138/3314003c-7995-4d96-b7cc-bba0f2b30351)

**Trello:**
https://trello.com/c/eUrMNKbx/629-update-links-based-on-your-organization

nnagewad commented:
Looking good so far. But will Fatality Notice, Field of Operation, and Sitewide Settings be hidden depending on your organization and permission?

farahTW commented:
> Looking good so far. But will Fatality Notice, Field of Operation, and Sitewide Settings be hidden depending on your organization and permission?

Yes It will show and hide based on organisation and user permission.
For MorePage it is already existing logic same as in legacy code.
For NewDocument Page we have implemented same as legacy logic.

farahTW commented:
**1.More Page screen shot when user's organisation is Ministry Of Defence:**
![image](https://github.com/alphagov/whitehall/assets/131259138/a406deed-2855-4031-bb2c-5f70161fd191)

2.More Page screen shot when user is GDS Editor and Organisation is GDS:

davidtrussler commented:
> > Looking good so far. But will Fatality Notice, Field of Operation, and Sitewide Settings be hidden depending on your organization and permission?
> 
> Yes It will show and hide based on organisation and user permission. For MorePage it is already existing logic same as in legacy code. For NewDocument Page we have implemented same as legacy logic.

Maybe I'm missing something but 
- I can't see where the logic is in the "New Document" page to exclude items based on organisation. 
- I don't think there is any logic at all along these lines in the "More" page

Also I'm not sure the intended logic is correct: shouldn't we be excluding some content on the basis of below?
- on the "New Document" page if the user is from an organisation other than GDS/MOD
- on the "More" page if the user is from an organisation other than GDS

farahTW commented:
> > > Looking good so far. But will Fatality Notice, Field of Operation, and Sitewide Settings be hidden depending on your organization and permission?
> > 
> > 
> > Yes It will show and hide based on organisation and user permission. For MorePage it is already existing logic same as in legacy code. For NewDocument Page we have implemented same as legacy logic.
> 
> Maybe I'm missing something but
> 
> * I can't see where the logic is in the "New Document" page to exclude items based on organisation.
> * I don't think there is any logic at all along these lines in the "More" page
> 
> Also I'm not sure the intended logic is correct: shouldn't we be excluding some content on the basis of below?
> 
> * on the "New Document" page if the user is from an organisation other than GDS/MOD
> * on the "More" page if the user is from an organisation other than GDS

@davidtrussler  it is the same logic as was in legacy implementation, we have added new document helper that handles the criteria

farahTW commented:
> > Looks good! just one thing to check with @nnagewad - the existing logic also removes the "Get involved" link (as well as "Fields of operation" and "Sitewide settings") on the "More" page for organisations other than GDS/MOD. Is this expected behaviour?
> 
> Good shout David. Looks like something I've missed as well.
> 
> @farahTW it looks like Get Involved is only present if your permission is set to GDS Editor.

Yes @nnagewad

davidtrussler commented:
> > Looks good! just one thing to check with @nnagewad - the existing logic also removes the "Get involved" link (as well as "Fields of operation" and "Sitewide settings") on the "More" page for organisations other than GDS/MOD. Is this expected behaviour?
> 
> Good shout David. Looks like something I've missed as well.
> 
> @farahTW it looks like Get Involved is only present if your permission is set to GDS Editor.

That looks like what we have - so will approve 👍

davidtrussler reviewed (approved):
Looks great, nice work! 👍
